packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -370,6 +370,34 @@ export class AzureCommunicationCallAdapter<AgentType extends CallAgent | BetaTea
       }
 
       this.context.updateClientState(clientState);
+
+      /* @conditional-compile-remove(call-transfer) */
+      const acceptedTransferCallState = this.context.getState().acceptedTransferCallState;
+
+      /* @conditional-compile-remove(call-transfer) */
+      // TODO: Remove this if statement when Calling SDK prevents accepting transfer requests that have timed out
+      // This is to handle the case when there has been an accepted transfer call that is now in the connected state
+      // AND is not the current call. Ensure we leave the current call.
+      if (
+        acceptedTransferCallState &&
+        acceptedTransferCallState.state === 'Connected' &&
+        acceptedTransferCallState.id !== this.call?.id
+      ) {
+        const cAgent = callAgent as CallAgent;
+        const transferCall = cAgent.calls.find((call) => (call as Call).id === acceptedTransferCallState.id);
+        if (transferCall) {
+          const oldCall = this.call;
+          this.processNewCall(transferCall);
+          // Do not hang up right away because we want the Transfer API to remove the call on a successful transfer. If
+          // we hang up right away the Transfer API will recognize the transfer as a failure even though we are now in
+          // the transfer call
+          setTimeout(() => {
+            if (oldCall?.state === 'Connected') {
+              oldCall.hangUp();
+            }
+          }, 2000);
```

Comment:
nit: can you add a comment why two seconds even just somethig like `Arbitrary wait time before hanging up. Time derived from manual testing.` just to highlight that 2 seconds was a number we came up with